src/scores/continuous/flip_flop_impl.py

```diff
@@ -348,3 +349,208 @@ def _encompassing_sector_size_np(
+            dimensions in the input data objects are collapsed.
+    Returns:
+        If `selections` are not supplied - An xarray.DataArray with dimensions
+        `dims` + 'threshold'. The DataArray is the proportion of the Flip-flop
+        Index calculated by collapsing dimension `sampling_dim` exceeding or
+        equal to `thresholds`.
+
+        If `selections` are supplied - An xarray.Dataset with dimensions `dims`
+        + 'threshold'. There is a data variable for each keyword in
+        `selections`, and corresponds to the Flip-Flop Index proportion
+        exceeding for the subset of data specified by the keyword values.
+
+    Examples:
+        >>> data = xr.DataArray(
+        ...     [[50, 20, 40, 80], [10, 50, 10, 100], [0, 30, 20, 50]],
+        ...     dims=['station_number', 'lead_day'],
+        ...     coords=[[10001, 10002, 10003], [1, 2, 3, 4]]
+        ... )
+
+        >>> flip_flop_index_proportion_exceeding(data, 'lead_day', [20])
+        <xarray.DataArray (threshold: 1)>
+        array([ 0.33333333])
+        Coordinates:
+          * threshold  (threshold) int64 20
+        Attributes:
+            sampling_dim: lead_day
+
+        >>> flip_flop_index_proportion_exceeding(
+        ...     data, 'lead_day', [20], days123=[1, 2, 3], all_days=[1, 2, 3, 4]
+        ... )
+        <xarray.Dataset>
+        Dimensions:    (threshold: 1)
+        Coordinates:
+          * threshold  (threshold) int64 20
+        Data variables:
+            days123    (threshold) float64 0.6667
+            all_days   (threshold) float64 0.3333
+        Attributes:
+            selections: {{'days123': [1, 2, 3], 'all_days': [1, 2, 3, 4]}}
+            sampling_dim: lead_day
+
+    See also:
+        `scores.continuous.flip_flop_index`
+
+    """
+    if (dims is not None) and (sampling_dim in list(dims)):
+        raise DimensionError(
+            f"`sampling_dim`: '{sampling_dim}' must not be in dimensions to preserve " f"`dims`: {list(dims)}"
+        )
+    # calculate the flip-flop index
+    flip_flop_data = flip_flop_index(data, sampling_dim, is_angular=is_angular, **selections)
+    # calculate the proportion exceeding each threshold
+    flip_flop_exceeding = proportion_exceeding(flip_flop_data, thresholds, dims=dims)
+    # overwrite the attributes
+    flip_flop_exceeding.attrs = flip_flop_data.attrs
+
+    return flip_flop_exceeding
+
+
+def proportion_exceeding(data: XarrayLike, thresholds: Iterable, dims: Optional[Iterable] = None):
+    """
+    Calculates the proportion of `data` equal to or exceeding `thresholds`.
+
+    Args:
+        data (xarray.Dataset or xarray.DataArray): The data from which
+            to calculate the proportion exceeding `thresholds`
+        thresholds (iterable): The proportion of Flip-Flop index results
+            equal to or exceeding these thresholds will be calculated.\
+            equal to or exceeding these thresholds will be calculated.\
+            the flip-flop index.
```

Comment:
This could be used more generically than just for the flip-flop index.  Could move into https://github.com/nci/scores/blob/develop/src/scores/processing.py